src/core/Global.ttslua

```diff
@@ -841,56 +845,55 @@ function applyOptionPanelChange(id, state)
-    spawnOrRemoveHelper(state, "Hand Helper", {-50.84, 1.6, 7.02}, {0, 270, 0}, "White")
-    spawnOrRemoveHelper(state, "Hand Helper", {-50.90, 1.6, -25.10}, {0, 270, 0}, "Orange")
-    spawnOrRemoveHelper(state, "Hand Helper", {-34.38, 1.6, 22.44}, {0, 000, 0}, "Green")
-    spawnOrRemoveHelper(state, "Hand Helper", {-16.69, 1.6, -22.42}, {0, 180, 0}, "Red")
+    optionPanel[id][1] = spawnOrRemoveHelper(state, "Hand Helper", {-50.84, 1.6, 7.02}, {0, 270, 0}, "White")
+    optionPanel[id][2] = spawnOrRemoveHelper(state, "Hand Helper", {-50.90, 1.6, -25.10}, {0, 270, 0}, "Orange")
+    optionPanel[id][3] = spawnOrRemoveHelper(state, "Hand Helper", {-34.38, 1.6, 22.44}, {0, 000, 0}, "Green")
+    optionPanel[id][4] = spawnOrRemoveHelper(state, "Hand Helper", {-16.69, 1.6, -22.42}, {0, 180, 0}, "Red")
 
   -- option: Show chaos bag manager
   elseif id == "showChaosBagManager" then
-    spawnOrRemoveHelper(state, "Chaos Bag Manager", {-67.8, 1.4, -49.5})
+    optionPanel[id] = spawnOrRemoveHelper(state, "Chaos Bag Manager", {-67.8, 1.4, -49.5})
 
   -- option: Show navigation overlay
   elseif id == "showNavigationOverlay" then
-    spawnOrRemoveHelper(state, "jaqenZann's Navigation Overlay", {-11.7, 1.4, -15})
+    optionPanel[id] = spawnOrRemoveHelper(state, "jaqenZann's Navigation Overlay", {-11.7, 1.4, -15})
   end
 end
 
 -- handler for spawn / remove functions of helper objects
 ---@param state Boolean Contains the state of the option: true = spawn it, false = remove it
 ---@param name String Name of the helper object
----@param position Vector Position of the object (where it will spawn or where it will be removed from)
+---@param position Vector Position of the object (where it will spawn)
 ---@param rotation Vector Rotation of the object for spawning (default: {0, 270, 0})
 ---@param color Color This is only needed for correctly setting the color of the "Hand Helper"
+-- returns the GUID of the spawnedObj (or nil if object was removed)
 function spawnOrRemoveHelper(state, name, position, rotation, color)
   if state then
-    spawnHelperObject(name, position, rotation, color)
-    Player["White"].pingTable(position)
+    Player.getPlayers()[1].pingTable(position)
+    return spawnHelperObject(name, position, rotation, color).getGUID()
   else
-    removeHelperObject(name, position)
+    return removeHelperObject(name)
   end
 end
 
 -- copies the specified tool (by name) from the barrel
 ---@param name String Name of the object that should be copied
 ---@param position Position Desired position of the object
 function spawnHelperObject(name, position, rotation, color)
-  if rotation == nil then rotation = {0, 270, 0} end
-
   for _, obj in ipairs(getObjectFromGUID(BARREL_GUID).getData().ContainedObjects) do
     if obj["Nickname"] == name then
-      spawnObjectData({
+      return spawnObjectData({
         data = obj,
         position = position,
-        rotation = rotation,
+        rotation = rotation or {0, 270, 0},
```

Comment:
The object's data should have a rotation to it, which we should use rather than hardcoding the 270.  If you just leave this off it will use the rotation defined in obj

Reply:
This is used for overriding the position for e.g. spawning a correctly rotated hand helper for each player

Reply:
Right - got that :)  Wasn't worried about using the passed rotation.  But if the rotation isn't provided, you're overriding it with 0,270,0 instead of the object's actual rotation when it went into the bag.  270 may not be the right "up" for some objects.

Not going to block on this, but please consider a cleanup.

Reply:
Alright, changing this in part 6